pkg/vpn/vpn.go

```diff
@@ -148,7 +159,30 @@ func getTailscaleInfo() (VPNInfo, error) {
 	ipv4Address, _ := util.GetFirst4String(tailscaleOutput.TailscaleIPs)
 	ipv6Address, _ := util.GetFirst6String(tailscaleOutput.TailscaleIPs)
 
-	return VPNInfo{IPv4Address: net.ParseIP(ipv4Address), IPv6Address: net.ParseIP(ipv6Address), NodeID: "", ProviderName: "tailscale", VPNInterface: tailscaleIf}, nil
+	return VPNInfo{BackendState: tailscaleOutput.BackendState, IPv4Address: net.ParseIP(ipv4Address), IPv6Address: net.ParseIP(ipv6Address), NodeID: "", ProviderName: "tailscale", VPNInterface: tailscaleIf}, nil
+}
+
+// get Tailscale advertised IP list
+func GetAdvertisedIP() (string, error) {
```

Comment:
The name indicates that this returns a singular IP, but it in fact returns a comma-separated string of all advertised routes. Return a list of string, or the actual type, whatever is easiest to work with on the caller side.
```suggestion
// get Tailscale advertised route list
func GetAdvertisedRoutes() ([]netip.Prefix, error) {
```